portfolio/src/main/webapp/index.html

```diff
@@ -131,11 +139,17 @@ <h3>This is a map of my hometown. The locations on the map are places I like to
                     </div>
                 </div>
                 <div>
-                    <img src="/images/diaFavicon.png">
+                    <img src="/images/diaFavicon.png" width=auto height="60px">
                     <p>
                         <a target="_blank" href="https://icons8.com/icons/set/diamond">Diamond icon</a> icon by <a target="_blank" href="https://icons8.com">Icons8</a>
                     </p>
                 </div>
+                <div>
+                    <a data-flickr-embed="true" href="https://www.flickr.com/photos/cynthiasmoot/38437425020/in/photostream/" title="Poké Bop Boba Tea">
+                    <img src="https://live.staticflickr.com/4609/38437425020_57b3aace76_s.jpg" width="auto" height="60px" alt="Poké Bop Boba Tea">
+                    <p>Boba image by Cynthia Smoot on Flickr</p></a>
+                    <script async src="//embedr.flickr.com/assets/client-code.js" charset="utf-8"></script>
```

Comment:
i'm curious what this script does? i'm not very familiar with front end development, but it seems strange to me that you would embed a script from another website in your own website, is this common? I could imagine many abuse vectors for this (like making your machine do crypto mining on behalf of flickr...), but perhaps there are safeguards I don't know about?

Reply:
I am not actually sure what the script does. A lot of these photo sharing websites have embed links that they like people to use to cite an image, and this section was something I copied and pasted. I have seen it mentioned in forum posts on Flickr and other sites, but I have not been able to find out what it is used for.

It can be deleted though, so I will just get rid of it.

Reply:
Soonil, I think that is a legitimate concern. I don't think there are many safeguards in place. I know that Chrome was taking steps to block cryptojacking ads, but I don't know if that extended to website scripts.

But to answer your question- yes, this is pretty common. It's common in several ways. 

First, you might import javascript libraries from CDNs. For example, cloudflare provides CDNs to import angularjs, so you don't have to host it your self, thereby improving website speed and reducing your bandwidth: https://cdnjs.com/libraries/angular.js/

Second, services that require client-side code will provide links to import those scripts. Flickr does that here (I assume to provide citations automatically?), but Google does a similar thing for Analytics. For example, you would add analytics to your site with a tag like

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-xxxxxxxxx"></script>

I think it really comes down to trust here. Do you trust Google and Flickr to not serve malicious javascript from the URL that you import? I don't think there are many safeguards, though.